### PR TITLE
added max min attribute

### PR DESCRIPTION
**Pull request by jeremycrisp**
added support for max and min attributes

### CODE REVIEW

File: experiments/golden-results/StratoX-summary.txt [hunk 1]
```diff
@@ -204,11 +204,6 @@ Error message: ATTRIBUTE_MACHINE_ROUNDING unsupported
 Nkind: N_Attribute_Reference
 --
 Occurs: 8 times
-Calling function: Do_Object_Declaration_Full
```
Comment (martin-cs) [inline experiments/golden-results/StratoX-summary.txt:207]: Why are there now only 7 of these?

File: experiments/golden-results/libkeccak-summary.txt [hunk 2]
```diff
@@ -183,11 +173,6 @@ Calling function: Do_Expression
 Error message: Index of string unsupported
 Nkind: N_Indexed_Component
 --
-Occurs: 14 times
```
Comment (martin-cs) [inline experiments/golden-results/libkeccak-summary.txt:186]: And none of these?

File: experiments/golden-results/muen-summary.txt [hunk 3]
```diff
@@ -203,11 +203,6 @@ Calling function: Do_Expression
 Error message: Unknown expression kind
 Nkind: N_Allocator
 --
-Occurs: 5 times
```
Comment (martin-cs) [inline experiments/golden-results/muen-summary.txt:206]: Also these?

File: testsuite/gnat2goto/tests/attribute_max_min/test.adb [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+   Val2 := 11;
+
+   pragma Assert (Integer'Max (Val1,Val2) = Val2);
+
+   pragma Assert (Integer'Min (Val1,Val2) = Val1);  
+
+   -- expect this one to fail
+   pragma Assert (Integer'Max (Val1,Val2) = Val1);
+
+end Test;
```
Comment (martin-cs) [inline testsuite/gnat2goto/tests/attribute_max_min/test.adb:18]: New line?

File: testsuite/gnat2goto/tests/attribute_max_min/test.adb [hunk 5]
```diff
@@ -0,0 +1,18 @@
+procedure Test is
+
+  Val1 : Integer;
+  Val2 : Integer;
+
+begin
+
```
Comment (martin-cs) [inline testsuite/gnat2goto/tests/attribute_max_min/test.adb:7]: Please can you add another test case with non-deterministic values as well, something like...

```
if Val1 != Val2 then
  if Integer'Max(Val1, Val2) = Val1 then
    pragma Assert(Integer'Min(Val1, Val2) = Val2);
  else
    pragma Assert(Integer'Min(Val1, Val2) = Val1);
  end if;
else
  pragma Assert(Integer'Max(Val1, Val2) = Integer'Min(Val1, Val2));
end if;
```

**Review by martin-cs — APPROVED** [review]
Approve subject to the comments being addressed.

File: experiments/golden-results/StratoX-summary.txt (hunk 1 above)
Comment (jeremycrisp) [reply experiments/golden-results/StratoX-summary.txt:207]: the missing Nkind was an attribute for 'Min'.It is now being processed correctly.

File: testsuite/gnat2goto/tests/attribute_max_min/test.adb (hunk 4 above)
Comment (jeremycrisp) [reply testsuite/gnat2goto/tests/attribute_max_min/test.adb]: corrected

File: testsuite/gnat2goto/tests/attribute_max_min/test.adb (hunk 5 above)
Comment (jeremycrisp) [reply testsuite/gnat2goto/tests/attribute_max_min/test.adb:7]: new test added

File: experiments/golden-results/libkeccak-summary.txt (hunk 2 above)
Comment (martin-cs) [reply experiments/golden-results/libkeccak-summary.txt:186]: Brilliant.  Thanks for investigating.

**Review by martin-cs — APPROVED** [review]
Thanks for the fixes.

**Pull request merged**